File: Medii de Proiectare si Programare/HW/MPP_CSharp/MPP_CSharp/Service/ServiceTrack.cs
using System;
using System.Collections.Generic;
using MPP_CSharp.Domain;
using MPP_CSharp.Repository;

namespace MPP_CSharp.Service
{
    public class ServiceTrack
    {
        private readonly IRepositoryTrack _repoDBTrack;

        public ServiceTrack(IRepositoryTrack repoDBTrack)
        {
            _repoDBTrack = repoDBTrack ?? throw new ArgumentNullException(nameof(repoDBTrack));
        }

        public Track? SearchTrackById(long idTrack)
        {
            return _repoDBTrack.Search(idTrack);
        }

        public IEnumerable<Track> GetAllTracks()
        {
            return _repoDBTrack.GetAll();
        }

        public void AddTrack(string name, int minimumAge, int maximumAge, int distance)
        {
            var track = new Track(name, minimumAge, maximumAge, distance);
            _repoDBTrack.Save(track);
        }

        public Track? FindTrackByName(string trackName)
        {
            return _repoDBTrack.FindTrackByName(trackName);
        }

        public IEnumerable<Track> GetTracksByAge(int minimumAge, int maximumAge)
        {
            return _repoDBTrack.GetTracksByAge(minimumAge, maximumAge);
        }

        public IEnumerable<TrackDTO> GetTrackDTOs()
        {
            return _repoDBTrack.GetTrackDTOs();
        }

        public IEnumerable<TrackDTO> GetTrackDTOsByAge(int minimumAge, int maximumAge)
        {
            return _repoDBTrack.GetTrackDTOsByAge(minimumAge, maximumAge);
        }
    }
}

File: Medii de Proiectare si Programare/HW/mpp_hw_copy/mpp_hw/src/main/java/mpp/com/Service/ServiceTrack.java
package mpp.com.Service;

import mpp.com.Domain.Track;
import mpp.com.Domain.TrackDTO;
import mpp.com.Repository.RepoDBTrack;
import mpp.com.Repository.RepositoryTrack;

import java.util.Optional;

public class ServiceTrack {

    RepositoryTrack repoDBTrack;

    public ServiceTrack(RepositoryTrack repoDBTrack) {
        this.repoDBTrack = repoDBTrack;
    }

    public Optional<Track> searchTrackById(Long idTrack) throws IllegalArgumentException {
        return repoDBTrack.findOne(idTrack);
    }

    public Iterable<Track> getAllTracks() {
        return repoDBTrack.findAll();
    }

    public void addTrack(String name, int minimumAge, int maximumAge, int distance) {
        var track = new Track(name, minimumAge, maximumAge, distance);
        repoDBTrack.save(track);
    }

    public Optional<Track> findTrackByName(String trackName) {
        return repoDBTrack.findTrackByName(trackName);
    }

    public Iterable<Track> getTracksByAge(int minimumAge, int maximumAge){
        return repoDBTrack.getTracksByAge(minimumAge, maximumAge);
    }

    public Iterable<TrackDTO> getTrackDTOs() {
        return repoDBTrack.getTrackDTOs();
    }
    public Iterable<TrackDTO> getTrackDTOsByAge(int minimumAge, int maximumAge) {
        return repoDBTrack.getTrackDTOsByAge(minimumAge, maximumAge);
    }
}
